lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -1149,83 +1142,98 @@ namespace Js
             searchArg = scriptContext->GetLibrary()->GetUndefinedDisplayString();
         }
 
-        char16 const * const inputStr = pThis->GetString();
+        const char16* const inputStr = pThis->GetString();
         const char16 * const searchStr = searchArg->GetString();
-        const int len = pThis->GetLength();
-        int position = len;
-        int const searchLen = searchArg->GetLength();
+
+        // 4. Let numPos be ? ToNumber(position). (If position is undefined, this step produces the value NaN.)
+        // 5. If numPos is NaN, let pos be +infinity; otherwise, let pos be ToInteger(numPos).
+        // 6. Let len be the number of elements in S.
+        // 7. Let start be min(max(pos, 0), len).
+
+        const charcount_t inputLen = pThis->GetLength();
+        const charcount_t searchLen = searchArg->GetLength();
+        charcount_t position = inputLen;
+        const char16* const searchLowerBound = inputStr;
 
         // Determine if the main string can't contain the search string by length
-        if ( searchLen > len )
+        if (searchLen > inputLen)
         {
             return JavascriptNumber::ToVar(-1, scriptContext);
         }
 
-        if(args.Info.Count > 2)
+        if (args.Info.Count > 2)
         {
             double pos = JavascriptConversion::ToNumber(args[2], scriptContext);
             if (!JavascriptNumber::IsNan(pos))
```

Comment:
alright, LGTM

---
In reply to: [87507596](https://github.com/Microsoft/ChakraCore/pull/1910#discussion_r87507596) [](ancestors = 87507596)